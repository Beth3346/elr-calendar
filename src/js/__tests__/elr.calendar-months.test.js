import $ from 'jquery'
import elrMonths from '../elr-calendar-months'
import elrTimeUtlities from 'elr-time-utilities'
import fs from 'fs'
import path from 'path'

// const elrMonths = elrCalendarMonths()
const elrTime = elrTimeUtlities()
let evts
let renderDate
let $cal

const calendarHtml = path.join(__dirname, '..', 'calendar.html')
const html = fs.readFileSync(calendarHtml).toString()

beforeEach(() => {
  evts = elrTime.holidays
  renderDate = {
    month: 2,
    date: 1,
    year: 2018
  }
  document.body.innerHTML = html
  $cal = $('.elr-calendar')
})

describe('renderMonth', () => {
  it('should display a month', () => {
    elrMonths.renderMonth(renderDate, $cal, evts)

    expect($cal.find('.elr-calendar-month-view').html()).toMatchSnapshot()
  })
  it('should show appropriate navigation', () => {
    elrMonths.renderMonth(renderDate, $cal, evts)

    expect($('.elr-calendar-year-prev').length).toBe(1)
    expect($('.elr-calendar-year-prev').text()).toBe('2017')

    expect($('.elr-calendar-year-next').length).toBe(1)
    expect($('.elr-calendar-year-next').text()).toBe('2019')

    expect($('.elr-calendar-month-prev').length).toBe(1)
    expect($('.elr-calendar-month-prev').text()).toBe('February')

    expect($('.elr-calendar-month-next').length).toBe(1)
    expect($('.elr-calendar-month-next').text()).toBe('April')

    expect($('.elr-calendar-week-prev').length).toBe(1)
    expect($('.elr-calendar-week-prev').css('display')).toBe('none')
    expect($('.elr-calendar-week-next').length).toBe(1)
    expect($('.elr-calendar-week-next').css('display')).toBe('none')

    expect($('.elr-calendar-date-prev').length).toBe(1)
    expect($('.elr-calendar-date-prev').css('display')).toBe('none')
    expect($('.elr-calendar-date-next').length).toBe(1)
    expect($('.elr-calendar-date-next').css('display')).toBe('none')
  })
  it('should create a heading', () => {
    elrMonths.renderMonth(renderDate, $cal, evts)

    expect($('.elr-calendar-header').text()).toBe('March 2018')
  })
})
