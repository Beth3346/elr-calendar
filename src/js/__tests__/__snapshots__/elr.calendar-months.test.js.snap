// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderMonth should display a month 1`] = `"<h1 class=\\"elr-calendar-header\\">March 2018</h1><div><div class=\\"elr-calendar-heading\\"><div class=\\"elr-calendar-heading-day\\">Sunday</div><div class=\\"elr-calendar-heading-day\\">Monday</div><div class=\\"elr-calendar-heading-day\\">Tuesday</div><div class=\\"elr-calendar-heading-day\\">Wednesday</div><div class=\\"elr-calendar-heading-day\\">Thursday</div><div class=\\"elr-calendar-heading-day\\">Friday</div><div class=\\"elr-calendar-heading-day\\">Saturday</div></div><div class=\\"elr-month\\"><div class=\\"elr-week odd-week\\" data-week=\\"9\\"><div id=\\"date1252018\\" class=\\"elr-cal-muted\\" data-month=\\"1\\" data-date=\\"25\\" data-year=\\"2018\\" data-day=\\"0\\">25</div><div id=\\"date1262018\\" class=\\"elr-cal-muted\\" data-month=\\"1\\" data-date=\\"26\\" data-year=\\"2018\\" data-day=\\"1\\">26</div><div id=\\"date1272018\\" class=\\"elr-cal-muted\\" data-month=\\"1\\" data-date=\\"27\\" data-year=\\"2018\\" data-day=\\"2\\">27</div><div id=\\"date1282018\\" class=\\"elr-cal-muted\\" data-month=\\"1\\" data-date=\\"28\\" data-year=\\"2018\\" data-day=\\"3\\">28</div><div id=\\"date212018\\" class=\\"elr-date\\" data-month=\\"2\\" data-date=\\"1\\" data-year=\\"2018\\" data-day=\\"4\\">1</div><div id=\\"date222018\\" class=\\"elr-date\\" data-month=\\"2\\" data-date=\\"2\\" data-year=\\"2018\\" data-day=\\"5\\">2</div><div id=\\"date232018\\" class=\\"elr-date elr-cal-weekend\\" data-month=\\"2\\" data-date=\\"3\\" data-year=\\"2018\\" data-day=\\"6\\">3</div></div><div class=\\"elr-week even-week\\" data-week=\\"10\\"><div id=\\"date242018\\" class=\\"elr-date elr-cal-weekend\\" data-month=\\"2\\" data-date=\\"4\\" data-year=\\"2018\\" data-day=\\"0\\">4</div><div id=\\"date252018\\" class=\\"elr-date\\" data-month=\\"2\\" data-date=\\"5\\" data-year=\\"2018\\" data-day=\\"1\\">5</div><div id=\\"date262018\\" class=\\"elr-date\\" data-month=\\"2\\" data-date=\\"6\\" data-year=\\"2018\\" data-day=\\"2\\">6</div><div id=\\"date272018\\" class=\\"elr-date\\" data-month=\\"2\\" data-date=\\"7\\" data-year=\\"2018\\" data-day=\\"3\\">7</div><div id=\\"date282018\\" class=\\"elr-date\\" data-month=\\"2\\" data-date=\\"8\\" data-year=\\"2018\\" data-day=\\"4\\">8</div><div id=\\"date292018\\" class=\\"elr-date\\" data-month=\\"2\\" data-date=\\"9\\" data-year=\\"2018\\" data-day=\\"5\\">9</div><div id=\\"date2102018\\" class=\\"elr-date elr-cal-weekend\\" data-month=\\"2\\" data-date=\\"10\\" data-year=\\"2018\\" data-day=\\"6\\">10</div></div><div class=\\"elr-week odd-week\\" data-week=\\"11\\"><div id=\\"date2112018\\" class=\\"elr-date elr-cal-weekend\\" data-month=\\"2\\" data-date=\\"11\\" data-year=\\"2018\\" data-day=\\"0\\">11</div><div id=\\"date2122018\\" class=\\"elr-date\\" data-month=\\"2\\" data-date=\\"12\\" data-year=\\"2018\\" data-day=\\"1\\">12</div><div id=\\"date2132018\\" class=\\"elr-date\\" data-month=\\"2\\" data-date=\\"13\\" data-year=\\"2018\\" data-day=\\"2\\">13</div><div id=\\"date2142018\\" class=\\"elr-date\\" data-month=\\"2\\" data-date=\\"14\\" data-year=\\"2018\\" data-day=\\"3\\">14</div><div id=\\"date2152018\\" class=\\"elr-date\\" data-month=\\"2\\" data-date=\\"15\\" data-year=\\"2018\\" data-day=\\"4\\">15</div><div id=\\"date2162018\\" class=\\"elr-date\\" data-month=\\"2\\" data-date=\\"16\\" data-year=\\"2018\\" data-day=\\"5\\">16</div><div id=\\"date2172018\\" class=\\"elr-date elr-cal-weekend\\" data-month=\\"2\\" data-date=\\"17\\" data-year=\\"2018\\" data-day=\\"6\\">17<ul class=\\"elr-event-list\\"><li><a href=\\"#\\" class=\\"elr-cal-holiday\\"><span class=\\"elr-event elr-all-day-event\\">St. Patrick's Day</span></a></li></ul></div></div><div class=\\"elr-week even-week\\" data-week=\\"12\\"><div id=\\"date2182018\\" class=\\"elr-date elr-cal-weekend\\" data-month=\\"2\\" data-date=\\"18\\" data-year=\\"2018\\" data-day=\\"0\\">18</div><div id=\\"date2192018\\" class=\\"elr-date\\" data-month=\\"2\\" data-date=\\"19\\" data-year=\\"2018\\" data-day=\\"1\\">19</div><div id=\\"date2202018\\" class=\\"elr-date\\" data-month=\\"2\\" data-date=\\"20\\" data-year=\\"2018\\" data-day=\\"2\\">20</div><div id=\\"date2212018\\" class=\\"elr-date\\" data-month=\\"2\\" data-date=\\"21\\" data-year=\\"2018\\" data-day=\\"3\\">21</div><div id=\\"date2222018\\" class=\\"elr-date\\" data-month=\\"2\\" data-date=\\"22\\" data-year=\\"2018\\" data-day=\\"4\\">22</div><div id=\\"date2232018\\" class=\\"elr-date\\" data-month=\\"2\\" data-date=\\"23\\" data-year=\\"2018\\" data-day=\\"5\\">23</div><div id=\\"date2242018\\" class=\\"elr-date elr-cal-weekend\\" data-month=\\"2\\" data-date=\\"24\\" data-year=\\"2018\\" data-day=\\"6\\">24</div></div><div class=\\"elr-week odd-week\\" data-week=\\"13\\"><div id=\\"date2252018\\" class=\\"elr-date elr-cal-weekend\\" data-month=\\"2\\" data-date=\\"25\\" data-year=\\"2018\\" data-day=\\"0\\">25</div><div id=\\"date2262018\\" class=\\"elr-date\\" data-month=\\"2\\" data-date=\\"26\\" data-year=\\"2018\\" data-day=\\"1\\">26</div><div id=\\"date2272018\\" class=\\"elr-date\\" data-month=\\"2\\" data-date=\\"27\\" data-year=\\"2018\\" data-day=\\"2\\">27</div><div id=\\"date2282018\\" class=\\"elr-date\\" data-month=\\"2\\" data-date=\\"28\\" data-year=\\"2018\\" data-day=\\"3\\">28</div><div id=\\"date2292018\\" class=\\"elr-date\\" data-month=\\"2\\" data-date=\\"29\\" data-year=\\"2018\\" data-day=\\"4\\">29</div><div id=\\"date2302018\\" class=\\"elr-date\\" data-month=\\"2\\" data-date=\\"30\\" data-year=\\"2018\\" data-day=\\"5\\">30</div><div id=\\"date2312018\\" class=\\"elr-date elr-cal-weekend\\" data-month=\\"2\\" data-date=\\"31\\" data-year=\\"2018\\" data-day=\\"6\\">31</div></div></div></div>"`;
